src/hooks/use-prompt-usage.tsx

```diff
@@ -29,13 +32,22 @@ export function usePromptUsage() {
 	return useContext( promptUsageContext );
 }
 
+const calculateDaysRemaining = ( quotaResetDate: string ): number => {
+	const resetDate = new Date( quotaResetDate );
```

Comment:
If `quotaResetDate` is an empty string, this function will return `NaN`. I think this case could happen if we can't fetch the prompt usage from the API, for instance when being offline. In this case, the message in the AI assistant tab will show:

**You've reached your <a>usage limit</a> for this month. Your limit will reset in NaN days.** 

Not sure how the AI assistant handles offline cases, as far as I've checked there's no visual change. One approach we could do is to remove the last sentence when we don't know the reset date:

**You've reached your <a>usage limit</a> for this month.**

WDYT?

Reply:
Hey hey thanks for the thorough review, much appreciated!

I think for the offline mode, we have a different design so this component won't be visible at all at the end, I think. Here is the issue: https://github.com/Automattic/dotcom-forge/issues/7818
When the app goes offline, we should display the offline message instead.

Reply:
> I think for the offline mode, we have a different design so this component won't be visible at all at the end, I think. Here is the issue: [Automattic/dotcom-forge#7818](https://github.com/Automattic/dotcom-forge/issues/7818)
> When the app goes offline, we should display the offline message instead.

Ah, thanks for sharing the issue. I see that when being offline this message won't be shown. Nevertheless, this won't cover the case if the API request fails. Nevertheless, this is likely to be an edge case because on a failure we won't either the prompt usage and the limit message won't be shown. That said, please feel free to omit this suggestion.

Reply:
Good point!

I think it is indeed more of an edge case but one thing that we can try to do is to default to some sort of message that does not include any data from the API e.g. "Your limit will reset next month" in case there is an error. What do you think?

Reply:
Yeah, sounds good. I like the idea of using a more generic message.

Reply:
I checked this further and I was not really able to reproduce the case when this would trigger. The way I tried to test it is to break the endpoint so that the request fails but I was not able to get the error. I think that it happens because  `userCanSendMessage` condition is also calculated based on the data fetched from the endpoint. So if the endpoint fails, that condition also fails and the usage limit view is not displayed at all. 

Generally, I would be okay to leave the code as is. We could add something like:

`const UsageLimitReached = () => {
	const { daysUntilReset: initialDaysUntilReset } = usePromptUsage();
	const [daysUntilReset, setDaysUntilReset] = useState<number | null>(initialDaysUntilReset);

	useEffect(() => {
		if (isNaN(initialDaysUntilReset)) {
			setDaysUntilReset(null);
		} else {
			setDaysUntilReset(initialDaysUntilReset);
		}
	}, [initialDaysUntilReset]);

	// Determine the reset message
	const resetMessage =
		daysUntilReset === null
			? __("You've reached your <a>usage limit</a> for this month. Your limit will reset next month.")
			: daysUntilReset <= 0
			? __("You've reached your <a>usage limit</a> for this month. Your limit will reset today.")
			: sprintf(
					_n(
						"You've reached your <a>usage limit</a> for this month. Your limit will reset in %s day.",
						"You've reached your <a>usage limit</a> for this month. Your limit will reset in %s days.",
						daysUntilReset
					),
					daysUntilReset
			  );

	return (
		<div className="text-center h-12 px-2 pt-4 text-a8c-gray-70">
			{createInterpolateElement(resetMessage, {
				a: <Button onClick={() => getIpcApi().showUserSettings()} variant="link" />,
			})}
		</div>
	);
};
`

But I am not seeing a real-life scenario that could trigger it because of that condition.

Reply:
Good point. Yeah, seems this case won't ever happen as the component won't be rendered 👍. Let's keep it as-is. Thanks for checking it @katinthehatsite 🙇 !